packages/@webex/plugin-cc/test/unit/spec/cc.ts

```diff
@@ -1,348 +1,312 @@
+
+jest.mock('../../../src/logger-proxy', () => ({
+  __esModule: true,
+  default: {
+    logger: {
+      log: jest.fn(),
+      error: jest.fn(),
+    },
+    initialize: jest.fn(),
+  },
+}));
+
+jest.mock('../../../src/services/config');
+jest.mock('../../../src/services/core/HttpRequest');
+jest.mock('../../../src/services/WebCallingService');
+jest.mock('../../../src/services');
+
+// Mock AgentConfig
+const mockAgentConfig = {
+  getAgentProfile: jest.fn(),
+};
+jest.mock('../../../src/features/Agentconfig', () => {
+  return jest.fn().mockImplementation(() => mockAgentConfig);
+});
 
 describe('webex.cc', () => {
   let webex;
-  let webSocketMock;
-  let eventEmitter;
-  let mockAgentProfile: AgentConfig;
+  let mockHttpRequest;
 
   beforeEach(() => {
     webex = new MockWebex({
       children: {
         cc: ContactCenter,
       },
-    });
+      logger: {
+        log: jest.fn(),
+        error: jest.fn(),
+      },
+      config: config,
+      once: jest.fn((event, callback) => callback()),
+    }) as unknown as WebexSDK;
 
-    // Ensure webex.internal.services is initialized correctly
-    webex.internal = {
-      ...webex.internal,
-      services: {
-        get: jest.fn().mockReturnValue('https://api.example.com/'),
+    mockHttpRequest = {
+      subscribeNotifications: jest.fn(),
+    };
+    webex.cc.httpRequest = mockHttpRequest;
+
+    // Mock Services instance
+    const mockServicesInstance = {
+      agent: {
+        stationLogin: jest.fn(),
+        logout: jest.fn(),
+        reload: jest.fn(),
       },
     };
+    (Services.getInstance as jest.Mock).mockReturnValue(mockServicesInstance);
+    webex.cc.services = mockServicesInstance;
+  });
 
-    // Manually mock the necessary methods
-    webSocketMock = {
-      on: jest.fn(),
-      subscribeAndConnect: jest.fn(),
-      disconnectWebSocket: jest.fn(),
-      off: jest.fn(),
-    };
-    webex.cc.webSocket = webSocketMock;
-    
-    // Mock event emitter
-    eventEmitter = new (require('events')).EventEmitter();
-    webSocketMock.on.mockImplementation((event, callback) => {
-      eventEmitter.on(event, callback);
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('cc.getDeviceId', () => {
+    it('should return dialNumber when loginOption is EXTENSION', () => {
+      const loginOption = LoginOption.EXTENSION;
+      const dialNumber = '12345';
+      const result = webex.cc['getDeviceId'](loginOption, dialNumber);
+      expect(result).toBe(dialNumber);
     });
 
-    global.WEBSOCKET_EVENT_TIMEOUT = 100; // Set to a smaller value for tests
+    it('should return dialNumber when loginOption is AGENT_DN', () => {
+      const loginOption = LoginOption.AGENT_DN;
+      const dialNumber = '12345';
+      const result = webex.cc['getDeviceId'](loginOption, dialNumber);
+      expect(result).toBe(dialNumber);
+    });
 
+    it('should return WEB_RTC_PREFIX + dialNumber for other loginOptions', () => {
```

Comment:
update the description to say prefix + agentId

Reply:
addresed